mongo/options/example_test.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) MongoDB, Inc. 2023-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package options_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"sync"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type CustomLogger struct {
+	io.Writer
+	mu sync.Mutex
+}
+
+func (logger *CustomLogger) Info(level int, msg string, _ ...interface{}) {
+	logger.mu.Lock()
+	defer logger.mu.Unlock()
+
+	logger.Write([]byte(fmt.Sprintf("level=%d msg=%s\n", level, msg)))
```

Comment:
Optional: Use `fmt.Fprintf` to write to the logger/writer.

```suggestion
	fmt.Fprintf(logger, "level=%d msg=%s\n", level, msg)
```